### PR TITLE
[Serve] Update Serve CLI and REST API behavior to use new config

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

#25333 enhances the Serve REST API and the Serve config to better support deployment graphs. This change updates the rest of the Serve REST API and the Serve CLI to use this enhanced config. The CLI updates include:

* `serve run`: takes in either the updated config schema or a `FunctionNode`/`ClassNode` import path.
* `serve deploy`: uses the updated config format.
* `serve config`: prints the most recently requested configuration instead of all configurations for all live deployments.
* `serve build`: outputs the updated config format.
* `serve delete`/`serve shutdown`: `serve delete` is renamed to `serve shutdown`, and the existing `serve shutdown` is deleted.

Unit tests in `test_serve_head.py` and `test_cli.py` are updated to assess the new CLI and REST API behavior with deployment graph-based configurations.

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This change updates tests in `test_serve_head.py` and `test_cli.py`.

### CODE REVIEW

File: dashboard/modules/serve/serve_head.py [hunk 1]
```diff
@@ -21,12 +21,12 @@ def __init__(self, dashboard_head):
     @routes.get("/api/serve/deployments/")
     @optional_utils.init_ray_and_catch_exceptions(connect_to_serve=True)
     async def get_all_deployments(self, req: Request) -> Response:
-        from ray.serve.api import list_deployments
-        from ray.serve.application import Application
+        from ray.serve.context import get_global_client
+
+        client = get_global_client(_override_controller_namespace="serve")
```
Comment (simon-mo) [inline dashboard/modules/serve/serve_head.py:26]: update here now namespace PR merged

File: python/ray/serve/client.py [hunk 2]
```diff
@@ -354,6 +354,11 @@ def list_deployments(self) -> Dict[str, Tuple[DeploymentInfo, str]]:
             for deployment_route in deployment_route_list.deployment_routes
         }
 
+    @_ensure_connected
+    def get_app_config(self) -> Dict:
+        """Returns the last requests Serve config."""
```
Comment (simon-mo) [inline python/ray/serve/client.py:359]: grammar

File: python/ray/serve/controller.py [hunk 3]
```diff
@@ -528,6 +528,18 @@ async def get_serve_status(self) -> bytes:
 
         return status_info.to_proto().SerializeToString()
 
+    def get_app_config(self) -> Dict:
+        checkpoint = self.kv_store.get(CONFIG_CHECKPOINT_KEY)
+        if checkpoint is None:
```
Comment (simon-mo) [inline python/ray/serve/controller.py:533]: should we throw here?

Comment (simon-mo) [reply python/ray/serve/controller.py:533]: or empty dict?

**Review by simon-mo — COMMENTED** [review]
lg overall

File: dashboard/modules/serve/serve_head.py (hunk 1 above)
Comment (shrekris-anyscale) [reply dashboard/modules/serve/serve_head.py:26]: Ah, thanks for the catch!

File: python/ray/serve/client.py (hunk 2 above)
Comment (shrekris-anyscale) [reply python/ray/serve/client.py:359]: Good catch!

**Review by simon-mo — APPROVED** [review]
LGTM overall. Again, let's please please try to stop relying on external repos with working dirs. Many of these tests should be able to run with local working dir, can you please try investigate that? (mayby start ray in $HOME)?

**Pull request merged**